### PR TITLE
Fix #158 : Search box for oData entitities to add

### LINKED ISSUE #158: Search box for oData entitities to add
ERP systems have large oData files. For development only a few entities are required.
Searching for the right data entity can be improved with a search box on top of the selection.
Additionally, adding all related tables should be selectable as a lot of related tables are added which are not required.

### PR DESCRIPTION
**Pull request by unchase**
Fix #158 :

search_text_box

Added the tests below:

To **SchemaTypesViewModelTests:**

- [x] `FillingSearchText_ShouldFilterSchemaTypes`.

To **OperationImportsViewModelTests:**

- [x] `FillingSearchText_ShouldFilterOperationImports`.

### CODE REVIEW

**Comment by unchase** [comment]
@habbes @marabooy @KenitoInc @ElizabethOkerio @gathogojr @paulodero ready for review

**Comment by UM001** [comment]
nice!

File: src/ViewModels/SchemaTypesViewModel.cs [hunk 1]
```diff
@@ -27,6 +27,29 @@ internal class SchemaTypesViewModel : ConnectedServiceWizardPage
...
+            {
+                _searchText = value;
+
+                this.OnPropertyChanged(nameof(SearchText));
+                this.OnPropertyChanged(nameof(FilteredSchemaTypes));
+            }
+        }
+
+        public IEnumerable<SchemaTypeModel> FilteredSchemaTypes
+        {
```
Comment (paulodero) [inline src/ViewModels/SchemaTypesViewModel.cs:44]: Could you document the public properties.

File: src/ViewModels/OperationImportsViewModel.cs [hunk 2]
```diff
@@ -41,6 +41,30 @@ public OperationImportsViewModel(UserSettings userSettings = null) : base()
...
+            set
+            {
+                _searchText = value;
+
+                this.OnPropertyChanged(nameof(SearchText));
+                this.OnPropertyChanged(nameof(FilteredOperationImports));
+            }
+        }
+
+        public IEnumerable<OperationImportModel> FilteredOperationImports
```
Comment (paulodero) [inline src/ViewModels/OperationImportsViewModel.cs:58]: Document the public properties.

File: src/ViewModels/OperationImportsViewModel.cs [hunk 3]
```diff
@@ -41,6 +41,30 @@ public OperationImportsViewModel(UserSettings userSettings = null) : base()
...
+        {
+            get
+            {
+                return SearchText == null
+                    ? OperationImports
+                    : OperationImports.Where(x => x.Name.StartsWith(SearchText, StringComparison.InvariantCultureIgnoreCase));
```
Comment (paulodero) [inline src/ViewModels/OperationImportsViewModel.cs:64]: Do you think it is necessary to do a fuzzy match here instead?

**Review by paulodero — CHANGES_REQUESTED** [review]
Public properties/methods need to be documented. You can as well include bound operations as part of search results if necessary.

File: src/ViewModels/OperationImportsViewModel.cs (hunk 3 above)
Comment (unchase) [reply src/ViewModels/OperationImportsViewModel.cs:64]: > Do you think it is necessary to do a fuzzy match here instead?

I think it is not necessary. But I can replace `StartsWith` with `Contains`.

File: src/ViewModels/OperationImportsViewModel.cs [hunk 4]
```diff
@@ -41,6 +41,33 @@ public OperationImportsViewModel(UserSettings userSettings = null) : base()
         }
 
         public IEnumerable<OperationImportModel> OperationImports { get; set; }
+
+        private string _searchText;
+        public string SearchText
```
Comment (paulodero) [inline src/ViewModels/OperationImportsViewModel.cs:46]: would be great if you can document this as well.

**Review by paulodero — APPROVED** [review]
You can document the undocumented public properties/methods then :shipit:

**Comment by unchase** [comment]
> You can document the undocumented public properties/methods then :shipit:

Done!

**Review by habbes — APPROVED** [review]
Looks good to me.

**Comment by unchase** [comment]
Can you tell me when you plan to merge the PR?

**Comment by habbes** [comment]
Hi @unchase, apologies for the delay. We have experienced some delays in transferring the extension to .Net Foundation on VS Marketplace. As soon as this process is finalized, we will publish a new release.

**Pull request merged**